source/Projectile.h

```diff
@@ -96,6 +97,11 @@ class Projectile : public Body {
 	// Get the distance that this projectile has traveled.
 	double DistanceTraveled() const;
 
+	// Once the projectile has come into contact with a phasing device, it
+	// will be decided if it should completely phase trough or make contact.
+	bool Phases(const Ship &ship) const;
+	void SetPhases(const Ship &ship);
```

Comment:
I'm not sure I understand why this is needed. The collision detection should check whether the ship is phased and lets projectiles go through, no need to modify the projectile no?

Reply:
this is so we dont check multiple times for no reason (imagine a swarm of projectiles in you getting checked every frame)
lemme optimize! xD

Reply:
OH it occured back to me, there's another reason
if you have a phasing of 0.5 50% of projectiles pass trough, but you need to keep track - or you'll just get hit in the end and phasing is only useful with a value of 1, which is op